pkg/controller/jobframework/reconciler.go

```diff
@@ -166,7 +166,24 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 		}
 	}
 
-	// 5. handle job is suspended.
+	// 5. handle eviction
+	if evCond := apimeta.FindStatusCondition(wl.Status.Conditions, kueue.WorkloadEvicted); evCond != nil && evCond.Status == metav1.ConditionTrue {
+		if !job.IsSuspended() {
+			log.V(6).Info("The job is not suspended, stop")
+			return ctrl.Result{}, r.stopJob(ctx, job, object, wl, evCond.Message)
+		}
+		if workload.IsAdmitted(wl) {
+			if !job.IsActive() {
+				log.V(6).Info("The job is no longer active, clear the workloads admission")
+				workload.UnsetAdmissionWithCondition(wl, "Pending", evCond.Message)
+				return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, wl, true)
+			}
+			//The job is suspended but active, nothing to do now.
```

Comment:
```suggestion
			// The job is suspended but active, nothing to do now.
```

Let's add an empty space since we always do that in other places :)